inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/resample_ref.cl

```diff
@@ -49,83 +91,210 @@ KERNEL (resample_gpu_ref)(__global INPUT0_TYPE* input,
+                            coords_sum[i] = max(-PADS_BEGIN[i], min(in_coords[i] + index[i] - 1, PADS_END[i] + in_size[i] - 1));
+                            if (coords_sum[i] < 0 || coords_sum[i] >= in_size[i])
+                                isOutOfBounds = true;
+                            coeff_prod *= cubic_coeff[i][index[i]];
+                        }
+                        if (!isOutOfBounds)
+                            interp_val += coeff_prod * input[FUNC_CALL(get_input_index)(coords_sum[0], coords_sum[1], coords_sum[2], coords_sum[3], coords_sum[4])];
+                    }
+                }
+            }
+        }
+    }
+
+#if HAS_FUSED_OPS
+    #define OF_ID (out_coords[1])
+    FUSED_OPS;
+    OUTPUT_TYPE res = FUSED_OPS_RESULT;
+#else // HAS_FUSED_OPS
+    OUTPUT_TYPE res = ACTIVATION(TO_OUTPUT_TYPE(interp_val), ACTIVATION_PARAMS);
+#endif // HAS_FUSED_OPS
+    output[FUNC_CALL(get_output_index)(out_coords[0], out_coords[1], out_coords[2], out_coords[3], out_coords[4])] = res;
+#elif defined(SAMPLE_TYPE_LINEAR_ONNX) // defined(SAMPLE_TYPE_NEAREST) && FEATURE_PACKED_MODE
+    const int ox = get_global_id(0);
+    const int oy = get_global_id(1);
+    const int feature = 0;
+    const int batch = get_global_id(2);
+    const int PADDED_Y = in_size[3] + PADS_BEGIN[3] + PADS_END[3];
+    const int PADDED_X = in_size[4] + PADS_BEGIN[4] + PADS_END[4];
+    const float ix = FUNC_CALL(get_original_coordinate)(ox, SCALES[4], out_size[4], PADDED_X);
+    const float iy = FUNC_CALL(get_original_coordinate)(oy, SCALES[3], out_size[3], PADDED_Y);
+
+#ifdef LEFTOVERS
+    if (ox >= OUTPUT_SIZE_X)
+        return;
 #endif
-    output[FUNC_CALL(get_output_index)(batch, feature, oz, oy, ox)] = res;
 
-#elif defined(SAMPLE_TYPE_INTERP)
+    float in_y = fmax(0, fmin(iy, PADDED_Y - 1));
+    float in_x = fmax(0, fmin(ix, PADDED_X - 1));
+    int in_y1 = min((int)in_y, PADDED_Y - 1);
+    int in_y2 = min(in_y1 + 1, PADDED_Y - 1);
+    int in_x1 = min((int)in_x, PADDED_X - 1);
+    int in_x2 = min(in_x1 + 1, PADDED_X - 1);
+
+    const ACCUMULATOR_TYPE dx1 = (in_x1 != in_x2) ? TO_ACCUMULATOR_TYPE(fabs(in_x - in_x1)) : 0.5f;
+    const ACCUMULATOR_TYPE dx2 = (in_x1 != in_x2) ? TO_ACCUMULATOR_TYPE(fabs(in_x - in_x2)) : 0.5f;
+    const ACCUMULATOR_TYPE dy1 = (in_y1 != in_y2) ? TO_ACCUMULATOR_TYPE(fabs(in_y - in_y1)) : 0.5f;
+    const ACCUMULATOR_TYPE dy2 = (in_y1 != in_y2) ? TO_ACCUMULATOR_TYPE(fabs(in_y - in_y2)) : 0.5f;
+    in_y1 -= PADS_BEGIN[3];
+    in_y2 -= PADS_BEGIN[3];
+    in_x1 -= PADS_BEGIN[4];
+    in_x2 -= PADS_BEGIN[4];
+
+    bool tlOutOfBounds = in_y1 < 0 || in_y1 >= in_size[3] || in_x1 < 0 || in_x1 >= in_size[4];
+    bool trOutOfBounds = in_y1 < 0 || in_y1 >= in_size[3] || in_x2 < 0 || in_x2 >= in_size[4];
+    bool blOutOfBounds = in_y2 < 0 || in_y2 >= in_size[3] || in_x1 < 0 || in_x1 >= in_size[4];
+    bool brOutOfBounds = in_y2 < 0 || in_y2 >= in_size[3] || in_x2 < 0 || in_x2 >= in_size[4];
```

Comment:
Maybe it make sense to add something like `ZERO_PADDINGS` macro to avoid not needed calculations in some cases?